### ISSUE TITLE
Feature: Specify permalink pattern in template layouts

### ISSUE THREAD

**Issue by kleinfreund** [11ty/eleventy#121]
I want to specify a permalink pattern in a template layout like this:

**layout.njk**:

```
---
permalink: "{{ page.fileSlug }}/index.html"
---
{{ layoutContent | safe }}
```

Then, templates in `./content` using the layout template from above (i.e. `layout.njk`) would all end up with transformed permalinks:

```
BEFORE: ./content/2017-09-03-example.md → _site/content/2017-09-03-example/index.html
AFTER:  ./content/2017-09-03-example.md → _site/example/index.html
```

This is different from the current behavior which only allows setting the permalink on regular templates but not layout templates (if I’m not mistaken). Currently, you would need to set the same permalink pattern (i.e. `permalink: "{{ page.fileSlug }}/index.html"`) on all templates to achieve the above.

**Comment by kleinfreund** [comment]
This appears to not be fixed. The document for test 1 below should produce the same permalink style as instructed in the directory-specific data file, but it doesn’t.

**`./_includes/layouts/content.njk`**:

```nunjucks
---
layout: layouts/base.njk
permalink: "{{ page.fileSlug }}/index.html"
---
```

**`./content/content.json`**:

```json
{
  "layout": "layouts/content.njk"
}
```

**`./content/2000-01-01-test-1.md`**:

```nunjucks
Test 1
```

→ produces `./_site/content/2000-01-01-test-1/index.html`

**`./content/2000-01-01-test-2.md`**:

```nunjucks
---
permalink: "{{ page.fileSlug }}/index.html"
---
Test 2
```

→ produces `./_site/test-1/index.html`

**Comment by kleinfreund** [comment]
However, it works when adding the instruction to the directory-specific data file like this:

```json
{
  "layout": "layouts/content.njk",
  "permalink": "{{ page.fileSlug }}/index.html"
}
```

Correct if I’m wrong, but both should be equivalent, right?

**Comment by zachleat** [comment]
_Hmmmmmm_

yes, I think so. Reopening.

**Comment by zachleat** [comment]
So there _were_ tests around this but not using template/dir data files—which was the edge case we were missing.

This will be included with 0.3.7.
